src/bin/rustfmt.rs

```diff
@@ -138,6 +152,32 @@ fn determine_operation<I>(opts: &Options, args: I) -> Operation
         return Operation::Help;
     }
 
+    // if no file argument is supplied, read from stdin
+    if matches.free.len() != 1 {
+
+        // make sure the write mode is plain or not set
+        // (the other options would require a file)
+        match matches.opt_str("write-mode") {
+            Some(mode) => {
+                match mode.parse() {
+                    Ok(WriteMode::Plain) => (),
+                    _ => return Operation::InvalidInput("Using stdin requires write-mode to be \
+                                                         plain"
+                                                            .into()),
+                }
+            }
+            _ => (),
+        }
```

Comment:
Can we just ignore the write mode and always use plain? Would allow a bit more flexibility